ruby/ext/google/protobuf_c/message.c

```diff
@@ -905,6 +905,33 @@ static VALUE Message_freeze(VALUE _self) {
   return _self;
 }
 
+/*
+ * call-seq:
+ *     Message.deep_freeze => self
+ *
+ * Deeep freezes the message object recursively.
+ */
+static VALUE Message_deep_freeze(VALUE _self) {
```

Comment:
Let's call it `internal_deep_freeze` just so it's unambiguous that it's intended to be internal-only.

It doesn't make it harder to call, but if someone does call it they will know that it's an API that could change.